### ISSUE TITLE
LinAlgError: Array must not containinfs or NaNs

### ISSUE THREAD

**Issue by yyfyyfzkyzky** [JonJala/mtag#194]
I've been able to run MTAG and get results without any problems before. Only when I use GCST009379 on the GWAS catalog as trait1, it gives me an error no matter which GWAS I use for trait2. The error is as follows:
  File "/home/tmj/miniconda3/envs/pgwas/lib/python2.7/site-packages/numpy/linalg/linalg.py", line 218, in _assertFinite
    raise LinAlgError("Array must not contain infs or NaNs")
But I went by what others have asked before, and the usual QC. I made sure that the 3 columns pval, z, and n of trait1 did not have NA or Inf, or character formats. But I am still reporting errors.
log.txt[GCST009379.ebi-a-GCST006414.log](https://github.com/JonJala/mtag/files/13330183/GCST009379.ebi-a-GCST006414.log)
GWAS summary data: http://diagram-consortium.org/downloads.html (T2D GWAS meta-analysis - Unadjusted for BMI Published in Mahajan et al (2018b))

Please what can I do to fix this problem？

**Comment by paturley** [comment]
For the trait, can you calculate some summary statistics for the pval, z,
and n columns (mean, median, min, max, number of observations)?

**Comment by paturley** [comment]
What's the length of the n and z columns? (Maybe it's in there somewhere,
but the format of this information is very hard to parse.)

**Comment by yyfyyfzkyzky** [comment]
> length(gcst9379$n)
[1] 15085062
> length(gcst9379$z)
[1] 15085062
> nrow(gcst9379)
[1] 15085062

**Comment by gaojinyang888** [comment]
> Hmm. Does it only have problems when it is trait 1 or also when it is trait
> 2?
> […](#)

Hello, in the recent analysis, I also encountered a similar problem. I would be most grateful if you could share the possible causes of this problem with us.

**Comment by gaojinyang888** [comment]
Calling ./mtag.py \
--n-min 0.0 \
--sumstats ../analysis/MTAG/GWAS/1.txt,../analysis/MTAG/GWAS/2.txt \
--out ../analysis/MTAG/result/1_2

2025/04/17/03:43:49 PM Beginning MTAG analysis...
2025/04/17/03:43:49 PM MTAG will use the Z column for analyses.
2025/04/17/03:44:16 PM Read in Trait 1 summary statistics (13367299 SNPs) from ../analysis/MTAG/GWAS/1.txt ...
2025/04/17/03:44:16 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
2025/04/17/03:44:16 PM Munging Trait 1  <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><
2025/04/17/03:44:16 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
2025/04/17/03:44:16 PM Interpreting column names as follows:
2025/04/17/03:44:16 PM snpid:	Variant ID (e.g., rs number)
n:	Sample size
a1:	a1, interpreted as ref allele for signed sumstat.
pval:	p-Value
a2:	a2, interpreted as non-ref allele for signed sumstat.
z:	Directional summary statistic as specified by --signed-sumstats.
se:	Standard errors of BETA coefficients

2025/04/17/03:44:16 PM Reading sumstats from provided DataFrame into memory 10000000 SNPs at a time.
2025/04/17/03:44:32 PM Read 13367299 SNPs from --sumstats file.
Removed 1307037 SNPs with missing values.
Removed 0 SNPs with INFO <= None.
Removed 0 SNPs with MAF <= 0.01.
Removed 0 SNPs with SE <0 or NaN values.
Removed 0 SNPs with out-of-bounds p-values.
Removed 0 variants that were not SNPs. Note: strand ambiguous SNPs were not dropped.
12060262 SNPs remain.
2025/04/17/03:44:38 PM Removed 674 SNPs with duplicated rs numbers (12059588 SNPs remain).
2025/04/17/03:44:39 PM Removed 0 SNPs with N < 0.0 (12059588 SNPs remain).
2025/04/17/03:46:41 PM Median value of SIGNED_SUMSTAT was 0.00928104188303, which seems sensible.
2025/04/17/03:46:41 PM Dropping snps with null values
2025/04/17/03:46:42 PM 
Metadata:
2025/04/17/03:46:43 PM Mean chi^2 = 1.078
2025/04/17/03:46:43 PM Lambda GC = 1.045
2025/04/17/03:46:43 PM Max chi^2 = 1373.873
2025/04/17/03:46:43 PM 2417 Genome-wide significant SNPs (some may have been removed by filtering).
2025/04/17/03:46:43 PM 
Conversion finished at Thu Apr 17 15:46:43 2025
2025/04/17/03:46:43 PM Total time elapsed: 2.0m:27.27s
2025/04/17/03:47:02 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
2025/04/17/03:47:02 PM Munging of Trait 1 complete. SNPs remaining:	 12060436
2025/04/17/03:47:02 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>

2025/04/17/03:47:20 PM Trait 1: Dropped 848 SNPs for duplicate values in the "snp_name" column
2025/04/17/03:47:45 PM Read in Trait 2 summary statistics (14498646 SNPs) from ../analysis/MTAG/GWAS/2.txt ...
2025/04/17/03:47:45 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
2025/04/17/03:47:45 PM Munging Trait 2  <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><
2025/04/17/03:47:45 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
2025/04/17/03:47:45 PM Interpreting column names as follows:
2025/04/17/03:47:45 PM snpid:	Variant ID (e.g., rs number)
n:	Sample size
a1:	a1, interpreted as ref allele for signed sumstat.
pval:	p-Value
a2:	a2, interpreted as non-ref allele for signed sumstat.
z:	Directional summary statistic as specified by --signed-sumstats.
se:	Standard errors of BETA coefficients

2025/04/17/03:47:46 PM Reading sumstats from provided DataFrame into memory 10000000 SNPs at a time.
2025/04/17/03:48:04 PM Read 14498646 SNPs from --sumstats file.
Removed 356271 SNPs with missing values.
Removed 0 SNPs with INFO <= None.
Removed 0 SNPs with MAF <= 0.01.
Removed 0 SNPs with SE <0 or NaN values.
Removed 0 SNPs with out-of-bounds p-values.
Removed 696609 variants that were not SNPs. Note: strand ambiguous SNPs were not dropped.
13445766 SNPs remain.
2025/04/17/03:48:13 PM Removed 0 SNPs with duplicated rs numbers (13445766 SNPs remain).
2025/04/17/03:48:15 PM Removed 0 SNPs with N < 0.0 (13445766 SNPs remain).
2025/04/17/03:50:32 PM Median value of SIGNED_SUMSTAT was 0.043381171644, which seems sensible.
2025/04/17/03:50:32 PM Dropping snps with null values
2025/04/17/03:50:33 PM 
Metadata:
2025/04/17/03:50:34 PM Mean chi^2 = 1.088
2025/04/17/03:50:34 PM Lambda GC = 1.078
2025/04/17/03:50:34 PM Max chi^2 = 413.135
2025/04/17/03:50:35 PM 1712 Genome-wide significant SNPs (some may have been removed by filtering).
2025/04/17/03:50:35 PM 
Conversion finished at Thu Apr 17 15:50:35 2025
2025/04/17/03:50:35 PM Total time elapsed: 2.0m:49.07s
2025/04/17/03:50:55 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
2025/04/17/03:50:55 PM Munging of Trait 2 complete. SNPs remaining:	 13445766
2025/04/17/03:50:55 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>

2025/04/17/03:51:23 PM Dropped 1865773 SNPs due to strand ambiguity, 10193815 SNPs remain in intersection after merging trait1
2025/04/17/03:51:47 PM Dropped 9725 SNPs due to inconsistent allele pairs from phenotype 2. 8669532 SNPs remain.
2025/04/17/03:51:53 PM Flipped the signs of of 1788608 SNPs to make them consistent with the effect allele orderings of the first trait.
2025/04/17/03:51:58 PM Dropped 0 SNPs due to strand ambiguity, 8669532 SNPs remain in intersection after merging trait2
2025/04/17/03:51:58 PM ... Merge of GWAS summary statistics complete. Number of SNPs:	 8669532
2025/04/17/03:52:21 PM Using 8669532 SNPs to estimate Omega (0 SNPs excluded due to strand ambiguity)
2025/04/17/03:52:21 PM Estimating sigma..
2025/04/17/03:53:34 PM Checking for positive definiteness ..
2025/04/17/03:53:34 PM Sigma hat:
[[1.092 0.006]
 [0.006 1.039]]
2025/04/17/03:53:34 PM Mean chi^2 of SNPs used to estimate Omega is low for some SNPsMTAG may not perform well in this situation.
2025/04/17/03:53:34 PM Beginning estimation of Omega ...
2025/04/17/03:53:34 PM Using GMM estimator of Omega ..
2025/04/17/03:53:35 PM Checking for positive definiteness ..
2025/04/17/03:53:35 PM Array must not contain infs or NaNs
Traceback (most recent call last):
  File "mtag.py", line 1577, in <module>
    mtag(args)
  File "mtag.py", line 1378, in mtag
    args.omega_hat = estimate_omega(args, Zs[not_SA], Ns[not_SA], args.sigma_hat)
  File "mtag.py", line 769, in estimate_omega
    return _posDef_adjustment(gmm_omega(Zs,Ns,sigma_LD))
  File "mtag.py", line 496, in _posDef_adjustment
    if is_pos_semidef(mat):
  File "mtag.py", line 495, in <lambda>
    is_pos_semidef = lambda m: np.all(np.linalg.eigvals(m) >= 0)
  File "/home/u2602634/.local/lib/python2.7/site-packages/numpy/linalg/linalg.py", line 1058, in eigvals
    _assertFinite(a)
  File "/home/u2602634/.local/lib/python2.7/site-packages/numpy/linalg/linalg.py", line 218, in _assertFinite
    raise LinAlgError("Array must not contain infs or NaNs")
LinAlgError: Array must not contain infs or NaNs
2025/04/17/03:53:35 PM Analysis terminated from error at Thu Apr 17 15:53:35 2025
2025/04/17/03:53:35 PM Total time elapsed: 9.0m:45.99s

**Comment by gaojinyang888** [comment]
> Can you provide the log file from your run?
> […](#)

Calling ./mtag.py
--n-min 0.0
--sumstats ../analysis/MTAG/GWAS/1.txt,../analysis/MTAG/GWAS/2.txt
--out ../analysis/MTAG/result/1_2

2025/04/17/03:43:49 PM Beginning MTAG analysis...
2025/04/17/03:43:49 PM MTAG will use the Z column for analyses.
2025/04/17/03:44:16 PM Read in Trait 1 summary statistics (13367299 SNPs) from ../analysis/MTAG/GWAS/1.txt ...
2025/04/17/03:44:16 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
2025/04/17/03:44:16 PM Munging Trait 1 <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><
2025/04/17/03:44:16 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
2025/04/17/03:44:16 PM Interpreting column names as follows:
2025/04/17/03:44:16 PM snpid: Variant ID (e.g., rs number)
n: Sample size
a1: a1, interpreted as ref allele for signed sumstat.
pval: p-Value
a2: a2, interpreted as non-ref allele for signed sumstat.
z: Directional summary statistic as specified by --signed-sumstats.
se: Standard errors of BETA coefficients

2025/04/17/03:44:16 PM Reading sumstats from provided DataFrame into memory 10000000 SNPs at a time.
2025/04/17/03:44:32 PM Read 13367299 SNPs from --sumstats file.
Removed 1307037 SNPs with missing values.
Removed 0 SNPs with INFO <= None.
Removed 0 SNPs with MAF <= 0.01.
Removed 0 SNPs with SE <0 or NaN values.
Removed 0 SNPs with out-of-bounds p-values.
Removed 0 variants that were not SNPs. Note: strand ambiguous SNPs were not dropped.
12060262 SNPs remain.
2025/04/17/03:44:38 PM Removed 674 SNPs with duplicated rs numbers (12059588 SNPs remain).
2025/04/17/03:44:39 PM Removed 0 SNPs with N < 0.0 (12059588 SNPs remain).
2025/04/17/03:46:41 PM Median value of SIGNED_SUMSTAT was 0.00928104188303, which seems sensible.
2025/04/17/03:46:41 PM Dropping snps with null values
2025/04/17/03:46:42 PM
Metadata:
2025/04/17/03:46:43 PM Mean chi^2 = 1.078
2025/04/17/03:46:43 PM Lambda GC = 1.045
2025/04/17/03:46:43 PM Max chi^2 = 1373.873
2025/04/17/03:46:43 PM 2417 Genome-wide significant SNPs (some may have been removed by filtering).
2025/04/17/03:46:43 PM
Conversion finished at Thu Apr 17 15:46:43 2025
2025/04/17/03:46:43 PM Total time elapsed: 2.0m:27.27s
2025/04/17/03:47:02 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
2025/04/17/03:47:02 PM Munging of Trait 1 complete. SNPs remaining: 12060436
2025/04/17/03:47:02 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>

2025/04/17/03:47:20 PM Trait 1: Dropped 848 SNPs for duplicate values in the "snp_name" column
2025/04/17/03:47:45 PM Read in Trait 2 summary statistics (14498646 SNPs) from ../analysis/MTAG/GWAS/2.txt ...
2025/04/17/03:47:45 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
2025/04/17/03:47:45 PM Munging Trait 2 <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><
2025/04/17/03:47:45 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
2025/04/17/03:47:45 PM Interpreting column names as follows:
2025/04/17/03:47:45 PM snpid: Variant ID (e.g., rs number)
n: Sample size
a1: a1, interpreted as ref allele for signed sumstat.
pval: p-Value
a2: a2, interpreted as non-ref allele for signed sumstat.
z: Directional summary statistic as specified by --signed-sumstats.
se: Standard errors of BETA coefficients

2025/04/17/03:47:46 PM Reading sumstats from provided DataFrame into memory 10000000 SNPs at a time.
2025/04/17/03:48:04 PM Read 14498646 SNPs from --sumstats file.
Removed 356271 SNPs with missing values.
Removed 0 SNPs with INFO <= None.
Removed 0 SNPs with MAF <= 0.01.
Removed 0 SNPs with SE <0 or NaN values.
Removed 0 SNPs with out-of-bounds p-values.
Removed 696609 variants that were not SNPs. Note: strand ambiguous SNPs were not dropped.
13445766 SNPs remain.
2025/04/17/03:48:13 PM Removed 0 SNPs with duplicated rs numbers (13445766 SNPs remain).
2025/04/17/03:48:15 PM Removed 0 SNPs with N < 0.0 (13445766 SNPs remain).
2025/04/17/03:50:32 PM Median value of SIGNED_SUMSTAT was 0.043381171644, which seems sensible.
2025/04/17/03:50:32 PM Dropping snps with null values
2025/04/17/03:50:33 PM
Metadata:
2025/04/17/03:50:34 PM Mean chi^2 = 1.088
2025/04/17/03:50:34 PM Lambda GC = 1.078
2025/04/17/03:50:34 PM Max chi^2 = 413.135
2025/04/17/03:50:35 PM 1712 Genome-wide significant SNPs (some may have been removed by filtering).
2025/04/17/03:50:35 PM
Conversion finished at Thu Apr 17 15:50:35 2025
2025/04/17/03:50:35 PM Total time elapsed: 2.0m:49.07s
2025/04/17/03:50:55 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
2025/04/17/03:50:55 PM Munging of Trait 2 complete. SNPs remaining: 13445766
2025/04/17/03:50:55 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>

2025/04/17/03:51:23 PM Dropped 1865773 SNPs due to strand ambiguity, 10193815 SNPs remain in intersection after merging trait1
2025/04/17/03:51:47 PM Dropped 9725 SNPs due to inconsistent allele pairs from phenotype 2. 8669532 SNPs remain.
2025/04/17/03:51:53 PM Flipped the signs of of 1788608 SNPs to make them consistent with the effect allele orderings of the first trait.
2025/04/17/03:51:58 PM Dropped 0 SNPs due to strand ambiguity, 8669532 SNPs remain in intersection after merging trait2
2025/04/17/03:51:58 PM ... Merge of GWAS summary statistics complete. Number of SNPs: 8669532
2025/04/17/03:52:21 PM Using 8669532 SNPs to estimate Omega (0 SNPs excluded due to strand ambiguity)
2025/04/17/03:52:21 PM Estimating sigma..
2025/04/17/03:53:34 PM Checking for positive definiteness ..
2025/04/17/03:53:34 PM Sigma hat:
[[1.092 0.006]
[0.006 1.039]]
2025/04/17/03:53:34 PM Mean chi^2 of SNPs used to estimate Omega is low for some SNPsMTAG may not perform well in this situation.
2025/04/17/03:53:34 PM Beginning estimation of Omega ...
2025/04/17/03:53:34 PM Using GMM estimator of Omega ..
2025/04/17/03:53:35 PM Checking for positive definiteness ..
2025/04/17/03:53:35 PM Array must not contain infs or NaNs
Traceback (most recent call last):
File "mtag.py", line 1577, in
mtag(args)
File "mtag.py", line 1378, in mtag
args.omega_hat = estimate_omega(args, Zs[not_SA], Ns[not_SA], args.sigma_hat)
File "mtag.py", line 769, in estimate_omega
return _posDef_adjustment(gmm_omega(Zs,Ns,sigma_LD))
File "mtag.py", line 496, in _posDef_adjustment
if is_pos_semidef(mat):
File "mtag.py", line 495, in
is_pos_semidef = lambda m: np.all(np.linalg.eigvals(m) >= 0)
File "/home/u2602634/.local/lib/python2.7/site-packages/numpy/linalg/linalg.py", line 1058, in eigvals
_assertFinite(a)
File "/home/u2602634/.local/lib/python2.7/site-packages/numpy/linalg/linalg.py", line 218, in _assertFinite
raise LinAlgError("Array must not contain infs or NaNs")
LinAlgError: Array must not contain infs or NaNs
2025/04/17/03:53:35 PM Analysis terminated from error at Thu Apr 17 15:53:35 2025
2025/04/17/03:53:35 PM Total time elapsed: 9.0m:45.99s

**Comment by JonJala** [comment]
It looks like the call to gmm_omega() on [line 769 of mtag.py](https://github.com/JonJala/mtag/blob/master/mtag.py#L769) is producing a matrix that contains infs or nans.  Unfortunately there are no logging statements in the code at that stage that would help narrow it down more.

It's not ideal, but you could replace line 769 with something like:

```
logging.info(f"Zs={Zs}")
logging.info(f"Ns={Ns}")
logging.info(f"sigma_LD={sigma_LD}")
unadj_omega = gmm_omega(Zs,Ns,sigma_LD)
logging.info(f"unadj_omega ={unadj_omega}")
return _posDef_adjustment(unadj_omega)

```

This would print out more information, and maybe one of those values would look funny and would provide more of an explanation.

**Comment by Licun1** [comment]
> It looks like the call to gmm_omega() on [line 769 of mtag.py](https://github.com/JonJala/mtag/blob/master/mtag.py#L769) is producing a matrix that contains infs or nans. Unfortunately there are no logging statements in the code at that stage that would help narrow it down more.
> 
> It's not ideal, but you could replace line 769 with something like:
> 
> ```
> logging.info(f"Zs={Zs}")
> logging.info(f"Ns={Ns}")
> logging.info(f"sigma_LD={sigma_LD}")
> unadj_omega = gmm_omega(Zs,Ns,sigma_LD)
> logging.info(f"unadj_omega ={unadj_omega}")
> return _posDef_adjustment(unadj_omega)
> ```
> 
> This would print out more information, and maybe one of those values would look funny and would provide more of an explanation.

Hello, I encounter a similar issue, which is also positioned at lien 769, at last I obtained only the .log file. I don't know how to solve this problem.
Calling ./mtag.py
--stream-stdout
--n-min 0.0
--sumstats ./GWAS_MTAG/1.txt,./GWAS_MTAG/2.txt
--out ./results_MTAG/1-2

2025/06/04/07:01:47 PM Beginning MTAG analysis...
2025/06/04/07:01:47 PM MTAG will use the Z column for analyses.
2025/06/04/07:02:13 PM Read in Trait 1 summary statistics (16853631 SNPs) from ./GWAS_MTAG/1.txt ...
2025/06/04/07:02:13 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
2025/06/04/07:02:13 PM Munging Trait 1 <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><
2025/06/04/07:02:13 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
2025/06/04/07:02:13 PM Interpreting column names as follows:
2025/06/04/07:02:13 PM snpid: Variant ID (e.g., rs number)
n: Sample size
a1: a1, interpreted as ref allele for signed sumstat.
pval: p-Value
a2: a2, interpreted as non-ref allele for signed sumstat.
z: Directional summary statistic as specified by --signed-sumstats.

2025/06/04/07:02:14 PM Reading sumstats from provided DataFrame into memory 10000000 SNPs at a time.
2025/06/04/07:02:48 PM Read 16853631 SNPs from --sumstats file.
Removed 0 SNPs with missing values.
Removed 0 SNPs with INFO <= None.
Removed 0 SNPs with MAF <= 0.01.
Removed 0 SNPs with SE <0 or NaN values.
Removed 0 SNPs with out-of-bounds p-values.
Removed 1 variants that were not SNPs. Note: strand ambiguous SNPs were not dropped.
16853630 SNPs remain.
2025/06/04/07:03:07 PM Removed 34191 SNPs with duplicated rs numbers (16819439 SNPs remain).
2025/06/04/07:03:10 PM Removed 0 SNPs with N < 0.0 (16819439 SNPs remain).
2025/06/04/07:03:39 PM Median value of SIGNED_SUMSTAT was 0.00538927687831, which seems sensible.
2025/06/04/07:03:40 PM Dropping snps with null values
2025/06/04/07:03:42 PM
Metadata:
2025/06/04/07:03:44 PM Mean chi^2 = 1.28
2025/06/04/07:03:45 PM Lambda GC = 1.186
2025/06/04/07:03:45 PM Max chi^2 = 199.484
2025/06/04/07:03:45 PM 9718 Genome-wide significant SNPs (some may have been removed by filtering).
2025/06/04/07:03:45 PM
Conversion finished at Wed Jun 4 19:03:45 2025
2025/06/04/07:03:45 PM Total time elapsed: 1.0m:32.33s
2025/06/04/07:04:18 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
2025/06/04/07:04:18 PM Munging of Trait 1 complete. SNPs remaining: 16853630
2025/06/04/07:04:18 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>

2025/06/04/07:05:01 PM Trait 1: Dropped 34191 SNPs for duplicate values in the "snp_name" column
2025/06/04/07:05:56 PM Read in Trait 2 summary statistics (32867901 SNPs) from ./GWAS_MTAG/2.txt ...
2025/06/04/07:05:56 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
2025/06/04/07:05:56 PM Munging Trait 2 <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><
2025/06/04/07:05:56 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
2025/06/04/07:05:56 PM Interpreting column names as follows:
2025/06/04/07:05:56 PM snpid: Variant ID (e.g., rs number)
n: Sample size
a1: a1, interpreted as ref allele for signed sumstat.
pval: p-Value
a2: a2, interpreted as non-ref allele for signed sumstat.
z: Directional summary statistic as specified by --signed-sumstats.

2025/06/04/07:05:58 PM Reading sumstats from provided DataFrame into memory 10000000 SNPs at a time.
2025/06/04/07:06:04 PM WARNING: 111 SNPs had P outside of (0,1]. The P column may be mislabeled.
2025/06/04/07:06:58 PM Read 32867901 SNPs from --sumstats file.
Removed 3074218 SNPs with missing values.
Removed 0 SNPs with INFO <= None.
Removed 0 SNPs with MAF <= 0.01.
Removed 0 SNPs with SE <0 or NaN values.
Removed 111 SNPs with out-of-bounds p-values.
Removed 929054 variants that were not SNPs. Note: strand ambiguous SNPs were not dropped.
28864518 SNPs remain.
2025/06/04/07:07:38 PM Removed 97268 SNPs with duplicated rs numbers (28767250 SNPs remain).
2025/06/04/07:07:44 PM Removed 0 SNPs with N < 0.0 (28767250 SNPs remain).
2025/06/04/07:08:29 PM Median value of SIGNED_SUMSTAT was -0.00535931790499, which seems sensible.
2025/06/04/07:08:30 PM Dropping snps with null values
2025/06/04/07:08:34 PM
Metadata:
2025/06/04/07:08:38 PM Mean chi^2 = 1.125
2025/06/04/07:08:40 PM Lambda GC = 1.088
2025/06/04/07:08:40 PM Max chi^2 = 1409.356
2025/06/04/07:08:40 PM 12467 Genome-wide significant SNPs (some may have been removed by filtering).
2025/06/04/07:08:40 PM
Conversion finished at Wed Jun 4 19:08:40 2025
2025/06/04/07:08:40 PM Total time elapsed: 2.0m:44.09s
2025/06/04/07:09:49 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
2025/06/04/07:09:49 PM Munging of Trait 2 complete. SNPs remaining: 28888262
2025/06/04/07:09:49 PM <><><<>><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>

2025/06/04/07:11:52 PM Trait 2: Dropped 121012 SNPs for duplicate values in the "snp_name" column
2025/06/04/07:12:05 PM Dropped 2661051 SNPs due to strand ambiguity, 14158388 SNPs remain in intersection after merging trait1
2025/06/04/07:13:22 PM Dropped 27368 SNPs due to inconsistent allele pairs from phenotype 2. 10776258 SNPs remain.
2025/06/04/07:13:28 PM Flipped the signs of of 3 SNPs to make them consistent with the effect allele orderings of the first trait.
2025/06/04/07:13:37 PM Dropped 0 SNPs due to strand ambiguity, 10776258 SNPs remain in intersection after merging trait2
2025/06/04/07:13:37 PM ... Merge of GWAS summary statistics complete. Number of SNPs: 10776258
2025/06/04/07:14:10 PM Using 10776258 SNPs to estimate Omega (0 SNPs excluded due to strand ambiguity)
2025/06/04/07:14:10 PM Estimating sigma..
2025/06/04/07:18:16 PM Checking for positive definiteness ..
2025/06/04/07:18:16 PM Sigma hat:
[[1.051 0.031]
[0.031 1.049]]
2025/06/04/07:18:17 PM Beginning estimation of Omega ...
2025/06/04/07:18:17 PM Using GMM estimator of Omega ..
2025/06/04/07:18:19 PM Checking for positive definiteness ..
2025/06/04/07:18:19 PM Array must not contain infs or NaNs
Traceback (most recent call last):
File "./mtag.py", line 1577, in
mtag(args)
File "./mtag.py", line 1378, in mtag
args.omega_hat = estimate_omega(args, Zs[not_SA], Ns[not_SA], args.sigma_hat)
File "./mtag.py", line 769, in estimate_omega
return _posDef_adjustment(gmm_omega(Zs,Ns,sigma_LD))
File "./mtag.py", line 496, in _posDef_adjustment
if is_pos_semidef(mat):
File "./mtag.py", line 495, in
is_pos_semidef = lambda m: np.all(np.linalg.eigvals(m) >= 0)
File "/home/licun/.local/lib/python2.7/site-packages/numpy/linalg/linalg.py", line 1058, in eigvals
_assertFinite(a)
File "/home/licun/.local/lib/python2.7/site-packages/numpy/linalg/linalg.py", line 218, in _assertFinite
raise LinAlgError("Array must not contain infs or NaNs")
LinAlgError: Array must not contain infs or NaNs
2025/06/04/07:18:19 PM Analysis terminated from error at Wed Jun 4 19:18:19 2025
2025/06/04/07:18:19 PM Total time elapsed: 16.0m:32.38s
